R/predict_functions.R

```diff
@@ -1,806 +1,804 @@
+  
+}
+
+# Immune Proportions #####
+
+calculate_immune_proportions <- function(immune_results) {
+  immune_proportions <- t(apply(immune_results,1,function(x){x/sum(x)}))
+  colnames(immune_proportions) <- paste0(colnames(immune_proportions)," Proportion")
+  return(immune_proportions)
+}
+
+
+# Grade predictor ##########
+
+predict_grade <- function(Data, grade_predictor,
+                          gene_id = c("ensembl_gene_id", "hgnc_symbol")[1],
+                          ...) {
+  
+  # Data must be a matrix in log2 transformed format, with sample as column and genes as rows
+  if (!class(Data)[1] %in% c("data.frame","matrix")) {
+    stop("Data must be in dataframe or matrix format.")
+  }
+  D <- Data
+  
+  if (class(grade_predictor)[1] != "rule_based_RandomForest") {
+    stop("Classifier must be a rule_based_RandomForest object.")
+  }
+  
+  if (!(gene_id %in% c("hgnc_symbol","ensembl_gene_id"))) {
+    stop("Gene ID must be one of the following: 'hgnc_symbol' or 'ensembl_gene_id'")
+  } else if (gene_id != "ensembl_gene_id") {
+    
+    # Testing
+    # load("D:/UROSCANSEQ_2024/Analysis/02.New_data/GradeClassifier/gene_info_grade_classifiers.RData")
+    # load("C:/Users/earam/LBCG/gene_info_lund.rda")
+    gene_info_lund <- LundTax2023Classifier::gene_info_lund
+    rownames(gene_info_lund) <- gene_info_lund[[gene_id]]
+    int_genes <- rownames(D)[which(rownames(D) %in% gene_info_lund[[gene_id]])]
+    rownames(D)[which(rownames(D) %in% gene_info_lund[[gene_id]])] <- gene_info_lund[int_genes,"ensembl_gene_id"]
+    
+  }
+  
+  
+  require(multiclassPairs)
+  grade_results <- predict_RF(classifier = grade_predictor, Data = D, ...)
+  
+  
+  
+  # if (ncol(D) == 1) cat(paste0("Prediction: ", grade_results$predictions_classes, "\n","Score: ", grade_results$predictions[,2], "\n"))
+  
+  return(grade_results)
+}
+
+
+# Function to calculate all scores #
+
+lund_scores <- function(Data, # Input data. Data must be a matrix in log2 transformed format, with sample as column and genes as rows
+                        gene_id = c("hgnc_symbol", "ensembl_gene_id")[2], # gene IDs
+                        scoring_method = c("ratio", "singscore")[1], # Method to calculate the proliferation score
+                        threshold_prostate = 4, # Gene expression threshold to flag a sample as possible prostate
+                        threshold_progression = 0.58, #  threshold to flag a sample as high risk of progression
+                        logTransform = TRUE, # Scores are calculated on log transformed data. If the data is already log transformed, set logTransformed to FALSE. If logTranform = TRUE, data will be log2 transformed (log2(data+1)) before calculating the scores
+                        adjust = FALSE,
+                        adj_factor = 5.1431,
+                        verbose = FALSE,
+                        ... # arguments to pass to the ranger functions (add impute = TRUE here if gene are missing)

```

Comment:
"ranger function", do you mean the `predict_RF` function?

Reply:
I took this from the multiclassPairs function, since predict_RF calls ranger inside and there is were the extra arguments will be used.